src/System.Linq/src/System/Linq/Skip.cs

```diff
@@ -121,5 +122,43 @@ private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSour
                 }
             }
         }
+
+#if netcoreapp11
+        public static IEnumerable<TSource> SkipLast<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+
+            if (count <= 0)
+            {
+                return source.Skip(0);
```

Comment:
The policy is that it's considered safe to return `source` if `source` comes from another linq operation, but not otherwise. `Skip` already has logic to determine that, so this can piggy-back. `Skip` is also naturally lighter than `SkipLast`, since it doesn't need a buffer queue, so this still wins in the case where it doesn't just return `source`.